### PR TITLE
Feature/delete section test question

### PR DESCRIPTION
**Pull request by arpita-20**

### CODE REVIEW

File: app/components/results/AttendedCandidateListItem.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const AttendedCandidateListItem = ({
     <div className="col-span-full">
       <div
         id="groupByTestRow"
-        className="col-span-full grid grid-cols-12 border-t-[1px] border-solid border-[#E5E7EB] bg-white px-12 py-4"
+        className="col-span-full grid grid-cols-12 border-t-[1px] border-solid border-[#E5E7EB] bg-white px-12 py-4 gap-3"
```
Comment (anuragP04) [inline app/components/results/AttendedCandidateListItem.tsx:20]: avoid using tailwind hardcoded values

File: app/components/results/AttendedCandidatesList.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const ExamAttendedCandidatesComponent = () => {
       {testData?.length !== 0 ? (
         <div className="bg-[#F9FAFB] pb-4  ">
           <div className="rounded-lg border border-solid border-[#E5E7EB] bg-tableHeader shadow-table">
```
Comment (anuragP04) [inline app/components/results/AttendedCandidatesList.tsx:18]: avoid using hardcoded values

File: app/components/results/AttendedCandidatesList.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ const ExamAttendedCandidatesComponent = () => {
       {testData?.length !== 0 ? (
```
Comment (anuragP04) [inline app/components/results/AttendedCandidatesList.tsx:16]: on line number 7, import interfaces as type, on line 7

File: app/components/sections/SectionCard.tsx [hunk 4]
```diff
@@ -9,13 +12,31 @@ const SectionCard = ({
...
+  actionStatusData,
 }: {
   name: string
   isActive: boolean
   questionsCount: number
   createdBy: string
-  createdAt: string
+  createdAt: Date
+  id: string
+  err: any
```
Comment (anuragP04) [inline app/components/sections/SectionCard.tsx:25]: remove type any

File: app/components/sections/SectionCard.tsx [hunk 5]
```diff
@@ -9,13 +12,31 @@ const SectionCard = ({
...
+  actionStatusData: any
 }) => {
+  // console.log(id, 'id')
```
Comment (anuragP04) [inline app/components/sections/SectionCard.tsx:28]: remove comment

File: app/components/sections/SectionCard.tsx [hunk 6]
```diff
@@ -9,13 +12,31 @@ const SectionCard = ({
...
+  const [isDelete, setIsDelete] = useState(false)
+  const submit = useSubmit()
+  const deleteSection = () => {
+    submit({ deleteSection: 'sectionDelete', id: id }, { method: 'post' })
+  }
+  useEffect(() => {
+    if (actionStatusData == 'Deleted Successfully..!') {
```
Comment (anuragP04) [inline app/components/sections/SectionCard.tsx:35]: use constants here

File: app/components/sections/SectionCard.tsx [hunk 7]
```diff
@@ -38,11 +59,46 @@ const SectionCard = ({
...
+                        active ? 'bg-primary text-white' : 'text-gray-900'
+                      } group flex w-full items-center rounded-md px-2 py-2 text-sm`}
+                      onClick={() => {
+                        setIsDelete(true)
+                      }}
+                    >
+                      {active ? (
+                        <Icon
+                          icon={'ic:outline-delete-outline'}
+                          className="mr-2 h-5 w-5 "
```
Comment (anuragP04) [inline app/components/sections/SectionCard.tsx:85]: remove extra space

File: app/components/sections/SectionCard.tsx [hunk 8]
```diff
@@ -38,11 +59,46 @@ const SectionCard = ({
...
+                          aria-hidden="true"
+                        />
+                      ) : (
+                        <Icon
+                          icon={'ic:outline-delete-outline'}
+                          className="mr-2 h-5 w-5 text-[#EF4444]"
```
Comment (anuragP04) [inline app/components/sections/SectionCard.tsx:91]: avoid using tailwind hardcoded values

File: app/components/sections/Sections.tsx [hunk 9]
```diff
@@ -5,13 +5,28 @@ import {} from '@remix-run/react'
...
+  actionStatusData: string | undefined
+  err: string | undefined
+  filter: string
+  setSelectedSection: (e: string) => void
+}) => {
   const path = `/sections/${section.id}`
   const location = useLocation() // to get current location
   const resolvedPath = useResolvedPath(path) // to get resolved path which would match with current location
-
+  const getSectionId = (id: string) => {
```
Comment (anuragP04) [inline app/components/sections/Sections.tsx:24]: what is this function doing, name is get, performing set operation
??

File: app/components/sections/Sections.tsx [hunk 10]
```diff
@@ -37,18 +55,25 @@ type SectionType = {
   filters: string
   setSortBy: (e: string) => void
   order: string
+  err: any
```
Comment (anuragP04) [inline app/components/sections/Sections.tsx:58]: remove any data type

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 11]
```diff
@@ -48,16 +48,16 @@ const AddQuestionInSection = () => {
   const transition = useTransition()
 
   const breadCrumbArray = [
+    // {
```
Comment (anuragP04) [inline app/components/sections/add-question/AddQuestionInSection.tsx:51]: remove comments if not needed in future

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 12]
```diff
@@ -48,16 +48,16 @@ const AddQuestionInSection = () => {
...
+    //   tabName: 'Section',
+    //   route: '/sections',
+    // },
     {
       tabName: 'Section',
```
Comment (anuragP04) [inline app/components/sections/add-question/AddQuestionInSection.tsx:56]: use constants here

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 13]
```diff
@@ -100,7 +100,9 @@ const AddQuestionInSection = () => {
         flag == 0 &&
         getQuestionType(selectedTypeOfQuestion) === QuestionTypes.multipleChoice
       ) {
-        toast.error('Select the Option', { toastId: 'correctOptionRequired' })
+        toast.error('Select the Correct Option', {
```
Comment (anuragP04) [inline app/components/sections/add-question/AddQuestionInSection.tsx:103]: use constants here

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 14]
```diff
@@ -109,7 +111,9 @@ const AddQuestionInSection = () => {
           QuestionTypes.singleChoice &&
         !singleChoiceAnswer
       ) {
-        toast.error('Select the Option', { toastId: 'correctOptionsRequired' })
+        toast.error('Select the Correct Option', {
```
Comment (anuragP04) [inline app/components/sections/add-question/AddQuestionInSection.tsx:114]: use constants here

File: app/components/sections/add-question/OptionForQuestion.tsx [hunk 15]
```diff
@@ -47,6 +47,9 @@ export default function OptionForQuestion({
         QuestionTypes.multipleChoice ||
       getQuestionType(selectedTypeOfQuestion) === QuestionTypes.singleChoice
     ) {
+      if (options.length > 9) {
+        return toast.error('you can add maximum ten options.', { toastId })
```
Comment (anuragP04) [inline app/components/sections/add-question/OptionForQuestion.tsx:51]: use constants here

File: app/routes/sections.tsx [hunk 16]
```diff
@@ -65,51 +73,80 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
-    .catch((err) => {
-      let title = 'Something went wrong..!'
-      if (err.code === 'P2002') {
-        title = 'Duplicate Title'
-      }
+    }
+    if (typeof description !== 'string' || description.length === 0) {
       return json<ActionData>(
-        { errors: { title, status: 400 } },
+        { errors: { title: 'description is required', status: 400 } },
```
Comment (anuragP04) [inline app/routes/sections.tsx:91]: use constants here

File: app/routes/sections.tsx [hunk 17]
```diff
@@ -65,51 +73,80 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
-  return section
+    }
+
+    let addHandle = null
+    await createSection({ name, description, createdById })
+      .then((res) => {
+        addHandle = json<ActionData>(
+          {
+            resp: {
+              status: 'Section Added Successfully..!',
```
Comment (anuragP04) [inline app/routes/sections.tsx:102]: use constants here

File: app/routes/sections.tsx [hunk 18]
```diff
@@ -65,51 +73,80 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
+              data: res as Section,
+              check: new Date(),
+            },
+          },
+          { status: 200 }
+        )
+      })
+
+      .catch((err) => {
+        let title = 'Something went wrong..!'
```
Comment (anuragP04) [inline app/routes/sections.tsx:112]: use constants everywhere in this file

File: app/routes/sections.tsx [hunk 19]
```diff
@@ -159,21 +196,26 @@ export default function SectionPage() {
...
       })
     }
-  }, [order, sortBy])
+  }, [order, sortBy, data.sections.length])
 
   useEffect(() => {
-    if (action) {
-      if (action.resp?.status === 200) {
+    if (sectionActionData) {
+      if (sectionActionData.resp?.status === 'Section Added Successfully..!') {
```
Comment (anuragP04) [inline app/routes/sections.tsx:203]: use constants here

File: cypress/e2e/section-tests/Section.cy.ts [hunk 20]
```diff
@@ -151,4 +152,44 @@ describe('Test for Section', () => {
...
+            .parent()
+            .within(() => {
+              cy.get('.verticalDots', { timeout: 6000 }).click()
+            })
+        }
+      })
+    })
+    cy.get('[data-cy="deleteSection"]').should('have.text', 'Delete').click()
+    cy.get('#delete-dialog').should('be.visible')
+    cy.get('.confirm-delete').should('have.text', cypress.delete).click()
```
Comment (richa-copods) [inline cypress/e2e/section-tests/Section.cy.ts:178]: Please use the actual constant instead of creating a new one

File: cypress/e2e/smoke.cy.ts [hunk 21]
```diff
@@ -92,6 +93,30 @@ describe('smoke tests', () => {
...
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#sections').should('have.text', cypress.Sections).click()
+    cy.get('#add-section').click()
+    cy.get('form > div')
+      .should('be.visible')
+      .within((el) => {
+        cy.get('input[placeholder="Enter Section Name"]').type(deleteSection)
```
Comment (richa-copods) [inline cypress/e2e/smoke.cy.ts:115]: Please use the constant for placeholder value

File: cypress/e2e/smoke.cy.ts [hunk 22]
```diff
@@ -92,6 +93,30 @@ describe('smoke tests', () => {
...
+        cy.get('textarea').type('Aptitude')
+        cy.get('[data-cy="submit"]').click()
+      })
+  })
   it('Add question in created section1', () => {
```
Comment (richa-copods) [inline cypress/e2e/smoke.cy.ts:120]: Please change it to- "Add question to a newly created section" or "Add question to the first section" as per requirement. Don't use `section1`

**Review by richa-copods — DISMISSED** [review]

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**